crates/air-lib/interpreter-data/src/value.rs

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::JValue;
+
+use serde::{Deserialize, Serialize};
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(transparent)]
+pub struct RawValue {
```

Comment:
Would it be better to implement it as enum with two variants: Parsed and Unparsed?

Reply:
That would require manual implementation of serde type, unfortunately.
One might use this type for `parsed` field, but it would be same as `Option` (though it is not bad).

Reply:
I think that serde implementation would be OK, but we can do it in separate PR.

Reply:
Something like `serde_with` might help to make it simple.
But I not sure that it worth it given `RawValue` is a type with interior mutability.  That mutable part is placed in the separate field.

Reply:
I think we can we can do smth like that:

```rust
struct RawValue {
  impl: RefCell<RawValueImpl>
}
```

I'd say that it'll make code more readable.

Reply:
Moreover it'll require less memory

Reply:
I think it is 4 words either way.